hydromt/validators/region.py

```diff
@@ -0,0 +1,239 @@
+"""Pydantic models for the validation of region specifications."""
```

Comment:
Looks good, but you can check out geojson_pydantic from developmentseed to get some inspiration / reuse their work if you want to deal with geometries.